kong/plugins/key-auth/handler.lua

```diff
@@ -131,11 +131,10 @@ local function do_authentication(conf)
 
   -- retrieve our consumer linked to this API key
 
-  local cache = kong.cache
-
   local credential_cache_key = kong.db.keyauth_credentials:cache_key(key)
-  local credential, err = cache:get(credential_cache_key, nil, load_credential,
-                                    key)
+  local credential, err      = kong.cache:get(credential_cache_key, nil,
```

Comment:
can't we use `kong.core_cache` here?

Reply:
this retrieves the credentials, not the consumer

Reply:
Credentials are plugins property. Not a core entity. So let's keep it like it is.

Reply:
The separation of caches isn't related to core vs plugin entities. It was introduced to ensure that cache keys derived from client-facing properties (e.g. request properties) can't churn out non client-facing cached items (i.e. with keys that aren't derived from client-facing properties).
Now this makes me concerned that over the last few months, we've treated the separated cache incorrectly. Has this been the reasoning behind choosing one or the other recently?